content/faq/libraries-frameworks/devops.mdx

```diff
@@ -51,8 +51,11 @@ showToc: true
 
 ## CI/CD
 
-- Azure DevOps Pipelines
-- Git Actions
+- [Azure DevOps Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/?view=azure-devops)
+- [Git Actions](https://docs.github.com/en/actions)
+- [Jenkins](https://github.com/jenkinsci/jenkins)
+- [GitLab CI](https://docs.gitlab.com/ee/ci/)
+- [Travis CI](https://docs.travis-ci.com/)
```

Comment:
Nice work.
BTW, since we are here could you please add CircleCi and AppVeyor?

Reply:
+1 TeamCity

Reply:
@aixasz Thank you, I completely forget TeamCity. LoL.